snuba/admin/clickhouse/system_queries.py

```diff
@@ -0,0 +1,132 @@
+from dataclasses import dataclass
+from typing import Any, Dict, Optional, Sequence, Tuple, Type, cast
+
+from snuba import settings
+from snuba.clickhouse.native import ClickhousePool
+from snuba.clusters.cluster import ClickhouseClientSettings
+from snuba.datasets.storages import StorageKey
+from snuba.datasets.storages.factory import get_storage
+from snuba.utils.serializable_exception import SerializableException
+
+
+class NonExistentSystemQuery(SerializableException):
+    pass
+
+
+class InvalidNodeError(SerializableException):
+    pass
+
+
+class InvalidStorageError(SerializableException):
+    pass
+
+
+class _QueryRegistry:
+    """Keep a mapping of SystemQueries to their names"""
+
+    def __init__(self) -> None:
+        self.__mapping: Dict[str, Type["SystemQuery"]] = {}
+
+    def register_class(self, cls: Type["SystemQuery"]) -> None:
+        existing_class = self.__mapping.get(cls.__name__)
+        if not existing_class:
+            self.__mapping[cls.__name__] = cls
+
+    def get_class_by_name(self, cls_name: str) -> Optional[Type["SystemQuery"]]:
+        return self.__mapping.get(cls_name)
+
+    @property
+    def all_queries(self) -> Sequence[Type["SystemQuery"]]:
+        return list(self.__mapping.values())
+
+
+_QUERY_REGISTRY = _QueryRegistry()
+
+
+@dataclass
+class SystemQuery:
+    sql: str
+
+    @classmethod
+    def to_json(cls) -> Dict[str, Optional[str]]:
+        return {
+            "sql": cls.sql,
+            "description": cls.__doc__,
+            "name": cls.__name__,
+        }
+
+    def __init_subclass__(cls) -> None:
+        _QUERY_REGISTRY.register_class(cls)
+        return super().__init_subclass__()
+
+    @classmethod
+    def from_name(cls, name: str) -> Optional[Type["SystemQuery"]]:
+        return _QUERY_REGISTRY.get_class_by_name(name)
+
+    @classmethod
+    def all_queries(cls) -> Sequence[Type["SystemQuery"]]:
+        return _QUERY_REGISTRY.all_queries
+
+
+class CurrentMerges(SystemQuery):
+    """Currently executing merges"""
+
+    sql = """
+        SELECT
+          count(),
+          is_currently_executing
+        FROM system.replication_queue
+        GROUP BY is_currently_executing
+        """
+
+
+class ActivePartitions(SystemQuery):
+    sql = """ SELECT
+        active,
+        count()
+    FROM system.parts
+    GROUP BY active
+    """
+
+
+def _is_valid_node(host: str, port: int) -> bool:
+    return (host == "localhost" or host == "127.0.0.1") and port == 9000
```

Comment:
I guess this is temporary. If we want to start with just one node, I'd suggest just validating it against the query node so it works in other environments not just dev.

Reply:
Are you assuming I would fetch the cluster by storage name and use https://github.com/getsentry/snuba/blob/ea463d9d35ebdcd6957fb0055d68fad06b2b85df/snuba/clusters/cluster.py#L184-L185 or something and check that it matches the request input host/port?